docs/docker-compose/start.sh

```diff
@@ -1,19 +1,52 @@
 #!/bin/bash
-# Clone paddles and teuthology
+set -e
+pushd teuthology
+if [ -z "$TEUTHOLOGY_BRANCH" -a -n "$GITHUB_HEAD_REF" ]; then
+    TEUTHOLOGY_BRANCH=${GITHUB_HEAD_REF}
+fi
+if [ ! -d ./teuthology ]; then
+    git clone \
+    --depth 1 \
+    -b ${TEUTHOLOGY_BRANCH:-$(git branch --show-current)} \
+    https://github.com/ceph/teuthology.git
+fi
```

Comment:
I completely agree that we should provide an easy way to use a fork. The more I think about this, though, the more it feels more sensible to use the current source tree as a volume inside the container, rather than managing a separate clone and building it into the image. Some nice things about that approach:

* Potentially less rebuild cycles (especially if we e.g. do bootstrap during build and store the virtualenv in the image and avoid re-bootstrapping unless requirements change)
* We get to drop some parameters and repo-managing code
* Avoids the frustrating situation where you have changes to your local clone that haven't yet been pushed, and you wonder why the bugfix/feature you're working on isn't behaving as you'd expect (because the changes aren't in the built image!)

If that sounds reasonable, maybe we leave this as-is and I can follow-up soon with some version of the above

Reply:
Agreed